app/src/main/java/de/tum/in/tumcampusapp/api/tumonline/interceptors/CheckErrorInterceptor.kt

```diff
@@ -0,0 +1,50 @@
+package de.tum.`in`.tumcampusapp.api.tumonline.interceptors
+
+import android.content.Context
+import com.tickaroo.tikxml.TikXml
+import de.tum.`in`.tumcampusapp.api.tumonline.exception.*
+import de.tum.`in`.tumcampusapp.api.tumonline.model.Error
+import de.tum.`in`.tumcampusapp.utils.Const
+import de.tum.`in`.tumcampusapp.utils.Utils
+import de.tum.`in`.tumcampusapp.utils.tryOrNull
+import okhttp3.Interceptor
+import okhttp3.Response
+
+class CheckErrorInterceptor(private val context: Context) : Interceptor {
+
+    private val tikXml = TikXml.Builder()
+            .exceptionOnUnreadXml(false)
+            .build()
+
+    @Throws(RequestLimitReachedException::class,
+            TokenLimitReachedException::class,
+            InvalidTokenException::class,
+            MissingPermissionException::class,
+            UnknownErrorException::class)
```

Comment:
If only there was a common exception type to be declared :thinking:

Reply:
🤔